packages/sdk/src/rpc/rpc.ts

```diff
@@ -0,0 +1,212 @@
+import { deserialize } from "@dao-xyz/borsh";
+import {
+  Proposal,
+  Proposals,
+  Query as QueryWasm,
+  Sdk as SdkWasm,
+  TransferToEthereum,
+} from "@namada/shared";
+
+import { SignedTx } from "../tx/types";
+import {
+  Balance,
+  BondsResponse,
+  DelegationTotals,
+  DelegatorsVotes,
+  GasCosts,
+  StakingPositions,
+  StakingTotals,
+  StakingTotalsResponse,
+  UnbondsResponse,
+} from "./types";
+
+/**
+ * API for executing RPC requests with Namada
+ */
+export class Rpc {
+  /**
+   * @param {SdkWasm} sdk - Instance of Sdk struct from wasm lib
+   * @param {QueryWasm} query - Instance of Query struct from wasm lib
+   */
+  constructor(
+    protected readonly sdk: SdkWasm,
+    protected readonly query: QueryWasm
+  ) {}
+
+  /**
+   * Query balances from chain
+   * @async
+   * @param {string} owner - Owner address
+   * @param {string[]} tokens - Array of token addresses
+   * @returns {Balance} [[tokenAddress, amount]]
+   */
+  async queryBalance(owner: string, tokens: string[]): Promise<Balance> {
+    return await this.query.query_balance(owner, tokens);
+  }
+
+  /**
+   * Query native token from chain
+   * @async
+   * @returns {string} Address of native token
+   */
+  async queryNativeToken(): Promise<string> {
+    return await this.query.query_native_token();
+  }
+
+  /**
+   * Query public key
+   * Return string of public key if it has been revealed on chain, otherwise, return null
+   * @async
+   * @param {string} address - Address to query
+   * @returns {string|null} String of public key if found
```

Comment:
This type doesn't match the TypeScript type. Actually, I thought eslint-plugin-jsdoc should catch that, so it might need to be configured to check against the TypeScript type (that would be really nice to have :smile:).

Reply:
So there is a typescript recommended ruleset which prevents you from adding type to the args/return values :D
So this:
```ts
  /**
   * Get address and public key from private key
   * @param {string} privateKey - Private key
   * @returns {Address} Address and public key
   */
  getAddress(privateKey: string): Address {
```
becomes this:
```ts
   * Get address and public key from private key
   * @param privateKey - Private key
   * @returns Address and public key
   */
  getAddress(privateKey: string): Address {
```

But this will make jsdoc2md not adding types to md :/ 

We can try to use https://typedoc.org/ instead though

Reply:
I quickly tried out typedoc and it looks nice. It will also warn us about classes etc. missing from the generated documentation.